### PR TITLE
Added new encoding options to WebP encoding

### PR DESCRIPTION
**Pull request by AaaK00**
Added more options for encoding WebP images.

### CODE REVIEW

**Comment by dreampiggy** [comment]
Thanks for your PR.

Seems like you add more options which libwebp supports. This sounds reasonable. I'll review later and merge to release a new version.

Tips: Did you check the libwebp's version ? Does all these APIs and options available in libwebp 1.0.0 ? Maybe we need to bump libwebp's dependency version.

**Comment by AaaK00** [comment]
> Seems like you add more options which libwebp supports. This sounds reasonable. I'll review later and merge to release a new version.

Yeap, this PR adds support for parameters specified here: https://developers.google.com/speed/webp/docs/api#advanced_encoding_api
> 
> Tips: Did you check the libwebp's version ? Does all these APIs and options available in libwebp 1.0.0 ? Maybe we need to bump libwebp's dependency version.

I believe all these options are supported in libwebp 1.0.0. But doesn't the project already use 1.1.0? At least that's specified in the swift.package. If not, then that's probably something that should be considered.

Also, one test is failing atm, but don't think that's related to this PR. At least for me it fails also when running tests locally on master branch.

**Comment by dreampiggy** [comment]
I'll fix the master branch. This PR looks fine for me.

File: SDWebImageWebPCoder/Classes/SDImageWebPCoder.h [hunk 1]
```diff
@@ -12,11 +12,29 @@
...
  Built in coder that supports WebP and animated WebP
  */
 @interface SDImageWebPCoder : NSObject <SDProgressiveImageCoder, SDAnimatedImageCoder>
 
 @property (nonatomic, class, readonly, nonnull) SDImageWebPCoder *sharedCoder;
 
+- (void) updateWebPOptionsToConfig:(WebPConfig * _Nonnull)config
+                       maxFileSize:(NSUInteger)maxFileSize
+                           options:(nullable SDImageCoderOptions *)options;
+
```
Comment (dreampiggy) [inline SDWebImageWebPCoder/Classes/SDImageWebPCoder.h:39]: One question, should this API be exposed to public ?


Actually, this WebP Coder is a top-level wrapper. So I don't want user to touch `WebPConfig` low level struct or something.

Can this be marked as private ? Users can just use the public API from `SDImageCoder` protocol. Like `decodedImageWithData:options:`

Comment (dreampiggy) [reply SDWebImageWebPCoder/Classes/SDImageWebPCoder.h:39]: If you want to call this API in Unit Test. One tip:

```objective-c
// Put these code into SDWebImageWebPCoderTests.m 


@interfae SDImageWebPCoder ()
- (void) updateWebPOptionsToConfig:(WebPConfig * _Nonnull)config
                       maxFileSize:(NSUInteger)maxFileSize
                           options:(nullable SDImageCoderOptions *)options;
@end

````

**Comment by dreampiggy** [comment]
master fix the test case, you can merge and try again :)

File: SDWebImageWebPCoder/Classes/SDImageWebPCoder.h (hunk 1 above)
Comment (AaaK00) [reply SDWebImageWebPCoder/Classes/SDImageWebPCoder.h:39]: Good point! 

Does not need to be public, just added it there to be able to test it in unit tests. (My Obj-c skills are bit rusty as I have mainly done Swift development for past three years. 😄 ) I'll try to do it in the way you suggested!

**Comment by AaaK00** [comment]
Rebased to master and removed the method updateWebPOptionsToConfig from the public API.

**Comment by dreampiggy** [comment]
I'll merge this, then fix little issues (licences comment, Xcode Project config, etc), and release new version .

Thanks for your contribution !

**Comment by dreampiggy** [comment]
Released [0.8.0](https://github.com/SDWebImage/SDWebImageWebPCoder/releases/tag/0.8.0)

**Pull request merged**